packages/mui-material/src/utils/getChildRef.js

```diff
@@ -0,0 +1,6 @@
+export default function getChildRef(children) {
+  // 'ref' is passed as prop in React 19, whereas 'ref' is directly attached to children in React 18
+  // below check is to ensure 'ref' is accessible in both cases
+  // return Object.keys(children.props).includes('ref') ? children.props.ref : children.ref;
+  return children.props.ref ?? children.ref;
```

Comment:
Depending on the types and if `children` is always defined or not, we should check for `children` being truthy. As it is right now, this will fail when passing `null`, `undefined`, string, array, etc. It will only work when passing a React element as single child.

Reply:
covered all possible cases where it might fail in this commit https://github.com/mui/material-ui/pull/42818/commits/1d5f95d9f53e5618cb434c54ee1cd2440f551630. hopefully didn't missed any